app/api/server/v1/voip/rooms.ts

```diff
@@ -7,7 +7,21 @@ import { API } from '../../api';
 import { VoipRoom, LivechatVisitors, Users } from '../../../../models/server/raw';
 import { LivechatVoip } from '../../../../../server/sdk';
 import { ILivechatAgent } from '../../../../../definition/ILivechatAgent';
+import { hasPermission } from '../../../../authorization/server';
+import { typedJsonParse } from '../../../../../lib/typedJSONParse';
 
+type DateParam = { start?: string; end?: string };
+const validateDateParams = (property: string, date: DateParam | string): DateParam => {
```

Comment:
This function is doing two things, which is a violation of the single-responsibility principle. Better split it into parsing and validation functions.

Reply:
Oh, that's a good thing :) didn't know we're following solid-esque principles now.

May I suggest to share that on the engineering channel instead of this PR? That way others can benefit too from new practices 😄 
Did we had a discussion about them 👀 ?